src/plugins/controls/public/control_group/actions/delete_control_action.tsx

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+
+import { EuiButtonIcon, EuiToolTip } from '@elastic/eui';
+import { ViewMode, isErrorEmbeddable } from '@kbn/embeddable-plugin/public';
+import { Action, IncompatibleActionError } from '@kbn/ui-actions-plugin/public';
+
+import { ACTION_DELETE_CONTROL } from '.';
+import { pluginServices } from '../../services';
+import { ControlGroupStrings } from '../control_group_strings';
+import { ControlEmbeddable, DataControlInput } from '../../types';
+import { isControlGroup } from '../embeddable/control_group_helpers';
+
+export interface DeleteControlActionContext {
+  embeddable: ControlEmbeddable<DataControlInput>;
+}
+
+export class DeleteControlAction implements Action<DeleteControlActionContext> {
+  public readonly type = ACTION_DELETE_CONTROL;
+  public readonly id = ACTION_DELETE_CONTROL;
+  public order = 2;
+
+  private openConfirm;
+
+  constructor() {
+    ({
+      overlays: { openConfirm: this.openConfirm },
+    } = pluginServices.getServices());
+  }
+
+  public readonly MenuItem = ({ context }: { context: DeleteControlActionContext }) => {
+    return (
+      <EuiToolTip content={this.getDisplayName(context)}>
+        <EuiButtonIcon
+          data-test-subj={`control-action-${context.embeddable.id}-delete`}
+          aria-label={this.getDisplayName(context)}
+          iconType={this.getIconType(context)}
+          onClick={() => this.execute(context)}
+          color="danger"
+        />
+      </EuiToolTip>
+    );
+  };
+
+  public getDisplayName({ embeddable }: DeleteControlActionContext) {
+    if (!embeddable.parent || !isControlGroup(embeddable.parent)) {
+      throw new IncompatibleActionError();
+    }
+    return ControlGroupStrings.floatingActions.getRemoveButtonTitle();
+  }
+
+  public getIconType({ embeddable }: DeleteControlActionContext) {
+    if (!embeddable.parent || !isControlGroup(embeddable.parent)) {
+      throw new IncompatibleActionError();
+    }
+    return 'cross';
+  }
+
+  public async isCompatible({ embeddable }: DeleteControlActionContext) {
+    if (isErrorEmbeddable(embeddable)) return false;
+    const controlGroup = embeddable.parent;
+    return Boolean(
+      !isErrorEmbeddable(embeddable) &&
```

Comment:
nit: Isn't this covered on line 67?

Reply:
Woop - it totally is. Caught that for `edit_control_action` but totally missed it here 🙃 Thanks!